### PR TITLE
Added data volumes

### PR DESCRIPTION
**Pull request by frankscholten**

### CODE REVIEW

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java [hunk 1]
```diff
@@ -70,6 +70,7 @@
                 .setContainer(Protos.ContainerInfo.newBuilder()
                         .setType(Protos.ContainerInfo.Type.DOCKER)
                         .setDocker(Protos.ContainerInfo.DockerInfo.newBuilder().setImage("mesos/elasticsearch-executor"))
+                        .addVolumes(Protos.Volume.newBuilder().setContainerPath("/data").setHostPath("/var/lib/elasticsearch").setMode(Protos.Volume.Mode.RW).build())
```
Comment (philwinder) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java]: Will mesos always have permission to RW /data externally? And will /var/lib/elasticsearch always exist?

File: system-test/build.gradle [hunk 2]
```diff
@@ -21,14 +21,14 @@ dependencies {
     compile 'com.github.docker-java:docker-java:1.3.0'
     compile 'com.mashape.unirest:unirest-java:1.4.5'
     compile 'com.jayway.awaitility:awaitility:1.6.3'
-    compile 'com.github.ContainerSolutions:mini-mesos:0.2.0'
+    compile 'com.github.ContainerSolutions:mini-mesos:c4ec773390329'
```
Comment (philwinder) [inline system-test/build.gradle]: Why is this necessary?

File: system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/DiscoverySystemTest.java [hunk 3]
```diff
@@ -25,6 +28,23 @@ public void testNodeDiscoveryRest() throws InterruptedException {
 
         ElasticsearchNodesResponse nodesResponse = new ElasticsearchNodesResponse(tasks, NODE_COUNT);
         assertTrue("Elasticsearch nodes did not discover each other within 5 minutes", nodesResponse.isDiscoverySuccessful());
-    }
+
```
Comment (philwinder) [inline system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/DiscoverySystemTest.java]: Is this testing that the data has actually been created _inside_ the container?
If so, this should be a seperate test, appropriately named. Probably not good to plonk it into the discovery test.

Comment (philwinder) [inline system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/DiscoverySystemTest.java]: Shouldn't there be a test to check whether the data is on the host machine?

Comment (philwinder) [inline system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/DiscoverySystemTest.java]: Shouldn't you clean out the data from the host machine after finishing the tests?

Comment (philwinder) [inline system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/DiscoverySystemTest.java]: Is it worth adding another reconcilliation test to test that if an executor dies and restarts, it continues to use the same data files, rather than creating new ones?

File: scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java (hunk 1 above)
Comment (frankscholten) [inline scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/TaskInfoFactory.java]: It will be created if it does not exist

File: system-test/build.gradle (hunk 2 above)
Comment (frankscholten) [inline system-test/build.gradle]: My tests failed earlier when I used 0.2.0. To debug the problem I added extra logging to AbstractContainer.remove() https://github.com/ContainerSolutions/mini-mesos/blob/master/src/main/java/org/apache/mesos/mini/container/AbstractContainer.java#L105

When I upgraded and re-ran the test it succeeded. Probably because of the recent commits after 0.2.0

File: system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/DiscoverySystemTest.java (hunk 3 above)
Comment (frankscholten) [inline system-test/src/systemTest/java/org/apache/mesos/elasticsearch/systemtest/DiscoverySystemTest.java]: I numbered your comments.

1. We don't test from inside the container. We only test the volume inside mesos-local.

2. That is tested here. The host machine is mesos-local

3. The volume is removed in https://github.com/ContainerSolutions/mini-mesos/blob/master/src/main/java/org/apache/mesos/mini/container/AbstractContainer.java#L105 

4. This is not supported yet. We would have to update the ClusterMonitor and the Scheduler.resourceOffers method to support this. The alternative is to use persistent volumes but those are blocked on authentication.

File: system-test/build.gradle (hunk 2 above)
Comment (philwinder) [inline system-test/build.gradle]: Ok cool, we should do another release of mini-mesos then.

Comment (frankscholten) [inline system-test/build.gradle]: Sounds good. 0.2.1
